packages/x-components/src/x-modules/search/components/results-list.vue

```diff
@@ -44,93 +43,122 @@
+
+    setup(props, { slots }) {
+      const xBus = useXBus();
+
+      const renderSlots = slots;
+      /**
+       * The {@link searchXModule | searchXModule } registered.
+       */
+      useRegisterXModule(searchXModule);
+
+      /**
+       * The results to render from the state.
+       *
+       * @remarks The results list are provided with `items` key. It can be
+       * concatenated with list items from components such as `BannersList`, `PromotedsList`,
+       * `BaseGrid` or any component that injects the list.
+       *
+       * @public
+       */
+      const items: ComputedRef<Result[]> = useState('search', ['results']).results;
+      provide<ComputedRef<Result[]>>(LIST_ITEMS_KEY as string, items);
+
+      /**
+       * The total number of results, taken from the state.
+       */
+      const totalResults: ComputedRef<number> = useState('search', ['totalResults']).totalResults;
+
+      /**
+       * It provides the search query.
+       * This query is updated only when the search request has succeeded.
+       */
+      let providedQuery = ref('');
+      provide<Ref<string>>(QUERY_KEY as string, providedQuery);
+
+      /**
+       * Indicates if there are more available results that have not been injected.
+       *
+       * @returns Boolean.
+       * @public
+       */
+      const hasMoreItems = computed<boolean>(() => {
+        return items.value.length < totalResults.value;
+      });
+      provide<ComputedRef<boolean>>(HAS_MORE_ITEMS_KEY as string, hasMoreItems);
+
+      /**
+       * The status of the search request, taken from the state.
+       */
+      const searchStatus: Ref<RequestStatus> = useState('search', ['status']).status;
+
+      /**
+       * The query of the search request, taken from the state.
+       */
+      const searchQuery: ComputedRef<string> = useState('search', ['query']).query;
+
+      /**
+       * Updates the query to be provided to the child components
+       * when the search request has succeeded.
+       *
+       * @param status - The status of the search request.
+       */
+      const updateQuery = (status: RequestStatus): void => {
+        if (status === 'success') {
+          providedQuery.value = searchQuery.value;
+        }
+      };
+
+      /**
+       * Watches the searchStatus and triggers updateQuery as callback
+       * when it changes.
+       *
+       * @param status - The status of the search request.
+       */
+      watch(searchStatus, () => updateQuery(searchStatus.value), { immediate: true });
+
+      /**
+       * It emits an {@link SearchXEvents.UserReachedResultsListEnd} event.
+       *
+       * @internal
+       */
+      const onInfiniteScrollEnd = (): void => {
+        xBus.emit('UserReachedResultsListEnd');
+      };
 
-    /**
-     * It emits an {@link SearchXEvents.UserReachedResultsListEnd} event.
-     *
-     * @internal
-     */
-    onInfiniteScrollEnd(): void {
-      this.$x.emit('UserReachedResultsListEnd');
+      return {
```

Comment:
There are too many things returned here which are not used in the template. So I think they can remove from here